{
 "matches": [
  {
   "vulnerability": {
    "id": "CVE-2007-4559",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2007-4559",
    "namespace": "nvd:cpe",
    "severity": "Medium",
    "urls": [
     "http://mail.python.org/pipermail/python-dev/2007-August/074290.html",
     "http://mail.python.org/pipermail/python-dev/2007-August/074292.html",
     "http://www.vupen.com/english/advisories/2007/3022",
     "https://bugzilla.redhat.com/show_bug.cgi?id=263261"
    ],
    "description": "Directory traversal vulnerability in the (1) extract and (2) extractall functions in the tarfile module in Python allows user-assisted remote attackers to overwrite arbitrary files via a .. (dot dot) sequence in filenames in a TAR archive, a related issue to CVE-2001-1267.",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:M/Au:N/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.8,
       "exploitabilityScore": 8.6,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python_software_foundation:python:3.10.6:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2007-4559",
      "versionConstraint": "none (unknown)",
      "cpes": [
       "cpe:2.3:a:python_software_foundation:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "python",
    "version": "3.10.6",
    "type": "binary",
    "locations": [
     {
      "path": "python3.10"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.10.6",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2015-20107",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2015-20107",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://bugs.python.org/issue24778",
     "https://github.com/python/cpython/issues/68966",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/46KWPTI72SSEOF53DOYQBQOCN4QQB2GE/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/53TQZFLS6O3FLIMVSXFEEPZSWLDZLBOX/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/57NECACX333A3BBZM2TR2VZ4ZE3UG3SN/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/5DBVY4YC2P6EPZZ2DROOXHDOWZ4BJFLW/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6QIKVSW3H6W2GQGDE5DTIWLGFNH6KKEW/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/AKGMYDVKI3XNM27B6I6RQ6QV3TVJAUCG/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ERYMM2QVDPOJLX4LYXWYIQN5FOIJLDRY/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/F3LNY2NHM6J22O6Q5ANOE3SZRK3OACKR/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FCIO2W4DUVVMI6L52QCC4TT2B3K5VWHS/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FIRUTX47BJD2HYJDLMI7JJBVCYFAPKAQ/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GPCLGZZJPVXFWUWVV5WCD5FNUAFLKBDN/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HAI2GBC7WKH7J5NH6J2IW5RT3VF2SF5M/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IFGV7P2PYFBMK32OKHCAC2ZPJQV5AUDF/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KAY6VBNVEFUXKJF37WFHYXUSRDEK34N3/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MYG3EMFR7ZHC46TDNM7SNWO64A3W7EUF/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ONXSGLASNLGFL57YU6WT6Y5YURSFV43U/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PTTZGLD2YBMMG6U6F5HOTPOGGPBIURMA/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UIOJUZ5JMEMGSKNISTOVI4PDP36FDL5Y/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/W5664BGZVTA46LQDNTYX5THG6CN4FYJX/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/WXF6MQ74HVIDDSR5AE2UDR24I6D4FEPC/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/XO2H6CKWLRGTTZCGUQVELW6LUH437Q3O/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/Y4E2WBEJ42CGLGDHD6ZXOLZ2W6G3YOVD/",
     "https://python-security.readthedocs.io/vuln/mailcap-shell-injection.html",
     "https://security.netapp.com/advisory/ntap-20220616-0001/"
    ],
    "description": "In Python (aka CPython) up to 3.10.8, the mailcap module does not add escape characters into commands discovered in the system mailcap file. This may allow attackers to inject shell commands into applications that call mailcap.findmatch with untrusted input (if they lack validation of user-provided filenames or arguments). The fix is also back-ported to 3.7, 3.8, 3.9",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:L/Au:S/C:P/I:C/A:P",
      "metrics": {
       "baseScore": 8,
       "exploitabilityScore": 8,
       "impactScore": 8.5
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:L/I:H/A:L",
      "metrics": {
       "baseScore": 7.6,
       "exploitabilityScore": 2.8,
       "impactScore": 4.7
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2015-20107",
      "versionConstraint": ">= 3.7.0, <= 3.7.15 || >= 3.8.0, <= 3.8.15 || >= 3.9.0, <= 3.9.15 || >= 3.10.0, < 3.10.8 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "python",
    "version": "3.10.6",
    "type": "binary",
    "locations": [
     {
      "path": "python3.10"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.10.6",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2018-1000500",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2018-1000500",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "http://lists.busybox.net/pipermail/busybox/2018-May/086462.html",
     "https://git.busybox.net/busybox/commit/?id=45fa3f18adf57ef9d743038743d9c90573aeeb91",
     "https://usn.ubuntu.com/4531-1/"
    ],
    "description": "Busybox contains a Missing SSL certificate validation vulnerability in The \"busybox wget\" applet that can result in arbitrary code execution. This attack appear to be exploitable via Simply download any file over HTTPS using \"busybox wget https://compromised-domain.com/important-file\".",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:M/Au:N/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.8,
       "exploitabilityScore": 8.6,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 8.1,
       "exploitabilityScore": 2.2,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2018-1000500",
      "versionConstraint": "< 1.32.0 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2020-10735",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2020-10735",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "http://www.openwall.com/lists/oss-security/2022/09/21/1",
     "http://www.openwall.com/lists/oss-security/2022/09/21/4",
     "https://access.redhat.com/security/cve/CVE-2020-10735",
     "https://bugzilla.redhat.com/show_bug.cgi?id=1834423",
     "https://docs.google.com/document/d/1KjuF_aXlzPUxTK4BMgezGJ2Pn7uevfX7g0_mvgHlL7Y",
     "https://github.com/python/cpython/issues/95778",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/2VCU6EVQDIXNCEDJUCTFIER2WVNNDTYZ/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/32AAQKABEKFCB5DDV5OONRZK6BS23HPW/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4EWKR2SPX3JORLWCXFY3KN2U5B5CIUQQ/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/6XL6E5A3I36TRR73VNBOXNIQP4AMZDFZ/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/76YE7AM37MRU76XJV4M27CWDAMUGNRYK/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/HSRPVJZL6DJFWKYRHMNJB7VCEUCBKRF5/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/IFGV7P2PYFBMK32OKHCAC2ZPJQV5AUDF/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/NHC6IUU7CLRQ3QLPWUXLONSG3SXFTR47/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/OKYE2DOI2X7WZXAWTQJZAXYIWM37HDCY/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/OT5U223OE5ZOUHZAZYSYSWVJQIKDE73E/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/OT5WQB7Z3CXOWVBD2AFAHYPA5ONYFFZ4/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/PD7FTLJOIGMUSCDR3JAN6WRFHJEE4PH5/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/SZYJSGLSCQOKXXFVJVJQAXLEOJBIWGEL/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/TD7JDDKJXK6D26XAN3YRFNM2LAJHT5UO/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/TMWPRAAJS7I6U3U45V7GZVXWNSECI22M/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/U4ZZV4CDFRMTPDBI7C5L43RFL3XLIGUY/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/UBPDVCDIUCEBE7C4NAGNA2KQJYOTPBAZ/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/V7ZUJDHK7KNG6SLIFXW7MNZ6O2PUJYK6/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/WXF6MQ74HVIDDSR5AE2UDR24I6D4FEPC/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/ZEOAJWGGY55QU35UM2OVZATBW5MX2OZD/"
    ],
    "description": "A flaw was found in python. In algorithms with quadratic time complexity using non-binary bases, when using int(\"text\"), a system could take 50ms to parse an int string with 100,000 digits and 5s for 1,000,000 digits (float, decimal, int.from_bytes(), and int() for binary bases 2, 4, 8, 16, and 32 are not affected). The highest threat from this vulnerability is to system availability.",
    "cvss": [
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
      "metrics": {
       "baseScore": 7.5,
       "exploitabilityScore": 3.9,
       "impactScore": 3.6
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2020-10735",
      "versionConstraint": ">= 3.7.0, < 3.7.14 || >= 3.8.0, < 3.8.14 || >= 3.9.0, < 3.9.14 || >= 3.10.0, < 3.10.7 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "python",
    "version": "3.10.6",
    "type": "binary",
    "locations": [
     {
      "path": "python3.10"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.10.6",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42374",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42374",
    "namespace": "nvd:cpe",
    "severity": "Medium",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "An out-of-bounds heap read in Busybox's unlzma applet leads to information leak and denial of service when crafted LZMA-compressed input is decompressed. This can be triggered by any applet/format that",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:L/AC:M/Au:N/C:P/I:N/A:P",
      "metrics": {
       "baseScore": 3.3,
       "exploitabilityScore": 3.4,
       "impactScore": 4.9
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:L/AC:H/PR:L/UI:N/S:U/C:L/I:N/A:H",
      "metrics": {
       "baseScore": 5.3,
       "exploitabilityScore": 1,
       "impactScore": 4.2
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42374",
      "versionConstraint": ">= 1.27.0, <= 1.33.1 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42376",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42376",
    "namespace": "nvd:cpe",
    "severity": "Medium",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "A NULL pointer dereference in Busybox's hush applet leads to denial of service when processing a crafted shell command, due to missing validation after a \\x03 delimiter character. This may be used for DoS under very rare conditions of filtered command input.",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:L/AC:M/Au:N/C:N/I:N/A:P",
      "metrics": {
       "baseScore": 1.9,
       "exploitabilityScore": 3.4,
       "impactScore": 2.9
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H",
      "metrics": {
       "baseScore": 5.5,
       "exploitabilityScore": 1.8,
       "impactScore": 3.6
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42376",
      "versionConstraint": ">= 1.16.0, < 1.34.0 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42378",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42378",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "A use-after-free in Busybox's awk applet leads to denial of service and possibly code execution when processing a crafted awk pattern in the getvar_i function",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:L/Au:S/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.5,
       "exploitabilityScore": 8,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 7.2,
       "exploitabilityScore": 1.2,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42378",
      "versionConstraint": ">= 1.16.0, <= 1.33.1 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42379",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42379",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "A use-after-free in Busybox's awk applet leads to denial of service and possibly code execution when processing a crafted awk pattern in the next_input_file function",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:L/Au:S/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.5,
       "exploitabilityScore": 8,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 7.2,
       "exploitabilityScore": 1.2,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42379",
      "versionConstraint": ">= 1.18.0, <= 1.33.1 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42380",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42380",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "A use-after-free in Busybox's awk applet leads to denial of service and possibly code execution when processing a crafted awk pattern in the clrvar function",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:L/Au:S/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.5,
       "exploitabilityScore": 8,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 7.2,
       "exploitabilityScore": 1.2,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42380",
      "versionConstraint": ">= 1.28.0, <= 1.33.1 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42381",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42381",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "A use-after-free in Busybox's awk applet leads to denial of service and possibly code execution when processing a crafted awk pattern in the hash_init function",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:L/Au:S/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.5,
       "exploitabilityScore": 8,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 7.2,
       "exploitabilityScore": 1.2,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42381",
      "versionConstraint": ">= 1.21.0, <= 1.33.1 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42382",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42382",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "A use-after-free in Busybox's awk applet leads to denial of service and possibly code execution when processing a crafted awk pattern in the getvar_s function",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:L/Au:S/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.5,
       "exploitabilityScore": 8,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 7.2,
       "exploitabilityScore": 1.2,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42382",
      "versionConstraint": ">= 1.26.0, <= 1.33.1 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42384",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42384",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "A use-after-free in Busybox's awk applet leads to denial of service and possibly code execution when processing a crafted awk pattern in the handle_special function",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:L/Au:S/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.5,
       "exploitabilityScore": 8,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 7.2,
       "exploitabilityScore": 1.2,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42384",
      "versionConstraint": ">= 1.18.0, <= 1.33.1 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42385",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42385",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "A use-after-free in Busybox's awk applet leads to denial of service and possibly code execution when processing a crafted awk pattern in the evaluate function",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:L/Au:S/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.5,
       "exploitabilityScore": 8,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 7.2,
       "exploitabilityScore": 1.2,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42385",
      "versionConstraint": ">= 1.16.0, <= 1.33.1 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2021-42386",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2021-42386",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://jfrog.com/blog/unboxing-busybox-14-new-vulnerabilities-uncovered-by-claroty-and-jfrog/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/6T2TURBYYJGBMQTTN2DSOAIQGP7WCPGV/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UQXGOGWBIYWOIVXJVRKHZR34UMEHQBXS/",
     "https://security.netapp.com/advisory/ntap-20211223-0002/"
    ],
    "description": "A use-after-free in Busybox's awk applet leads to denial of service and possibly code execution when processing a crafted awk pattern in the nvalloc function",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:L/Au:S/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.5,
       "exploitabilityScore": 8,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 7.2,
       "exploitabilityScore": 1.2,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2021-42386",
      "versionConstraint": ">= 1.16.0, <= 1.33.1 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2022-28391",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2022-28391",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://git.alpinelinux.org/aports/plain/main/busybox/0001-libbb-sockaddr2str-ensure-only-printable-characters-.patch",
     "https://git.alpinelinux.org/aports/plain/main/busybox/0002-nslookup-sanitize-all-printed-strings-with-printable.patch",
     "https://gitlab.alpinelinux.org/alpine/aports/-/issues/13661"
    ],
    "description": "BusyBox through 1.35.0 allows remote attackers to execute arbitrary code if netstat is used to print a DNS PTR record's value to a VT compatible terminal. Alternatively, the attacker could choose to change the terminal's colors.",
    "cvss": [
     {
      "version": "2.0",
      "vector": "AV:N/AC:M/Au:N/C:P/I:P/A:P",
      "metrics": {
       "baseScore": 6.8,
       "exploitabilityScore": 8.6,
       "impactScore": 6.4
      },
      "vendorMetadata": {}
     },
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 8.8,
       "exploitabilityScore": 2.8,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2022-28391",
      "versionConstraint": "<= 1.35.0 (unknown)",
      "cpes": [
       "cpe:2.3:a:busybox:busybox:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "busybox",
    "version": "1.30.1",
    "type": "binary",
    "locations": [
     {
      "path": "busybox"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*",
     "cpe:2.3:a:busybox:busybox:1.30.1:*:*:*:*:*:*:*"
    ],
    "purl": "",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2022-37454",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2022-37454",
    "namespace": "nvd:cpe",
    "severity": "Critical",
    "urls": [
     "https://csrc.nist.gov/projects/hash-functions/sha-3-project",
     "https://eprint.iacr.org/2023/331",
     "https://github.com/XKCP/XKCP/security/advisories/GHSA-6w4m-2xhg-2658",
     "https://lists.debian.org/debian-lts-announce/2022/10/msg00041.html",
     "https://lists.debian.org/debian-lts-announce/2022/11/msg00000.html",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/3ALQ6BDDPX5HU5YBQOBMDVAA2TSGDKIJ/",
     "https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/CMIEXLMTW5GO36HTFFWIPB3OHZXCT3G4/",
     "https://mouha.be/sha-3-buffer-overflow/",
     "https://news.ycombinator.com/item?id=33281106",
     "https://news.ycombinator.com/item?id=35050307",
     "https://www.debian.org/security/2022/dsa-5267",
     "https://www.debian.org/security/2022/dsa-5269"
    ],
    "description": "The Keccak XKCP SHA-3 reference implementation before fdc6fef has an integer overflow and resultant buffer overflow that allows attackers to execute arbitrary code or eliminate expected cryptographic properties. This occurs in the sponge function interface.",
    "cvss": [
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 9.8,
       "exploitabilityScore": 3.9,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2022-37454",
      "versionConstraint": ">= 3.6.0, < 3.7.16 || >= 3.8.0, < 3.8.16 || >= 3.9.0, < 3.9.16 || >= 3.10.0, < 3.10.9 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "python",
    "version": "3.10.6",
    "type": "binary",
    "locations": [
     {
      "path": "python3.10"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.10.6",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2022-42919",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2022-42919",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://github.com/python/cpython/compare/v3.10.8...v3.10.9",
     "https://github.com/python/cpython/compare/v3.9.15...v3.9.16",
     "https://github.com/python/cpython/issues/97514",
     "https://github.com/python/cpython/issues/97514#issuecomment-1310277840",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/FKGCQPIVHEAIJ77R3RSNSQWYBUDVWDKU/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/P2LHWWEI5OBQ6RELULMVU6KMDYG4WZXH/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PI5DYIED6U26BGX5IRZWNCP6TY4M2ZGZ/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/QLUGZSEAO3MBWGKCUSMKQIRYJZKJCIOB/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/R6KGIRHSENZ4QAB234Z36HVIDTRJ3MFI/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RDK3ZZBRYFO47ET3N4BNTKVXN47U6ICY/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VCRKBB5Y5EWTJUNC7LK665WO64DDXSTN/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/XX6LLAXGZVZ327REY6MDZRMMP47LJ53P/",
     "https://security.netapp.com/advisory/ntap-20221209-0006/"
    ],
    "description": "Python 3.9.x before 3.9.16 and 3.10.x before 3.10.9 on Linux allows local privilege escalation in a non-default configuration. The Python multiprocessing library, when used with the forkserver start method on Linux, allows pickles to be deserialized from any user in the same machine local network namespace, which in many system configurations means any user on the same machine. Pickles can execute arbitrary code. Thus, this allows for local user privilege escalation to the user that any forkserver process is running as. Setting multiprocessing.util.abstract_sockets_supported to False is a workaround. The forkserver start method for multiprocessing is not the default start method. This issue is Linux specific because only Linux supports abstract namespace sockets. CPython before 3.9 does not make use of Linux abstract namespace sockets by default. Support for users manually specifying an abstract namespace socket was added as a bugfix in 3.7.8 and 3.8.3, but users would need to make specific uncommon API calls in order to do that in CPython before 3.9.",
    "cvss": [
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H",
      "metrics": {
       "baseScore": 7.8,
       "exploitabilityScore": 1.8,
       "impactScore": 5.9
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2022-42919",
      "versionConstraint": ">= 3.7.3, <= 3.7.15 || >= 3.8.3, <= 3.8.15 || >= 3.9.0, < 3.9.16 || >= 3.10.0, < 3.10.9 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "python",
    "version": "3.10.6",
    "type": "binary",
    "locations": [
     {
      "path": "python3.10"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.10.6",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2022-45061",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2022-45061",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://github.com/python/cpython/issues/98433",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/2AOUKI72ACV6CHY2QUFO6VK2DNMVJ2MB/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/35YDIWCUMWTMDBWFRAVENFH6BLB65D6S/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/4WBZJNSALFGMPYTINIF57HAAK46U72WQ/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/63FS6VHY4DCS74HBTEINUDOECQ2X6ZCH/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/7WQPHKGNXUJC3TC3BDW5RKGROWRJVSFR/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/B3YI6JYARWU6GULWOHNUROSACT54XFFS/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GTPVDZDATRQFE6KAT6B4BQIQ4GRHIIIJ/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/IN26PWZTYG6IF3APLRXQJBVACQHZUPT2/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/JCDJXNBHWXNYUTOEV4H2HCFSRKV3SYL3/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/JTYVESWVBPD57ZJC35G5722Q6TS37WSB/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/KNE4GMD45RGC2HWUAAIGTDHT5VJ2E4O4/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/LKWAMPURWUV3DCCT4J7VHRF4NT2CFVBR/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/O67LRHDTJWH544KXB6KY4HMHQLYDXFPK/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ORVCQGJCCAVLN4DJDTWGREFCUWXKQRML/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/PLQ2BNZVBBAQPV3SPRU24ZD37UYJJS7W/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/QLUGZSEAO3MBWGKCUSMKQIRYJZKJCIOB/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RDK3ZZBRYFO47ET3N4BNTKVXN47U6ICY/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RH57BNT4VQERGEJ5SXNXSVMDYP66YD4H/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/RTN2OOLKYTG34DODUEJGT5MLC2PFGPBA/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/T3D5TX4TDJPXHXD2QICKTY3OCQC3JARP/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/UHVW73QZJMHA4MK7JBT7CXX7XSNYQEGF/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/VCMDX6IFKLOA3NXUQEV524L5LHTPI2JI/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/X3EJ6J7PXVQOULBQZQGBXCXY6LFF6LZD/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/XXZJL3CNAFS5PAIR7K4RL62S3Y7THR7O/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/YPNWZKXPKTNHS5FVMN7UQZ2UPCSEFJUK/",
     "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ZB5YCMIRVX35RUB6XPOWKENCVCJEVDRK/",
     "https://security.netapp.com/advisory/ntap-20221209-0007/"
    ],
    "description": "An issue was discovered in Python before 3.11.1. An unnecessary quadratic algorithm exists in one path when processing some inputs to the IDNA (RFC 3490) decoder, such that a crafted, unreasonably long name being presented to the decoder could lead to a CPU denial of service. Hostnames are often supplied by remote servers that could be controlled by a malicious actor; in such a scenario, they could trigger excessive CPU consumption on the client attempting to make use of an attacker-supplied supposed hostname. For example, the attack payload could be placed in the Location header of an HTTP response with status code 302. A fix is planned in 3.11.1, 3.10.9, 3.9.16, 3.8.16, and 3.7.16.",
    "cvss": [
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
      "metrics": {
       "baseScore": 7.5,
       "exploitabilityScore": 3.9,
       "impactScore": 3.6
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2022-45061",
      "versionConstraint": "<= 3.7.15 || >= 3.8.0, <= 3.8.15 || >= 3.9.0, <= 3.9.15 || >= 3.10.0, <= 3.10.8 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 || = 3.11.0 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "python",
    "version": "3.10.6",
    "type": "binary",
    "locations": [
     {
      "path": "python3.10"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.10.6",
    "upstreams": []
   }
  },
  {
   "vulnerability": {
    "id": "CVE-2023-24329",
    "dataSource": "https://nvd.nist.gov/vuln/detail/CVE-2023-24329",
    "namespace": "nvd:cpe",
    "severity": "High",
    "urls": [
     "https://github.com/python/cpython/pull/99421",
     "https://pointernull.com/security/python-url-parse-problem.html"
    ],
    "description": "An issue in the urllib.parse component of Python before v3.11 allows attackers to bypass blocklisting methods by supplying a URL that starts with blank characters.",
    "cvss": [
     {
      "version": "3.1",
      "vector": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:H/A:N",
      "metrics": {
       "baseScore": 7.5,
       "exploitabilityScore": 3.9,
       "impactScore": 3.6
      },
      "vendorMetadata": {}
     }
    ],
    "fix": {
     "versions": [],
     "state": "unknown"
    },
    "advisories": []
   },
   "relatedVulnerabilities": [],
   "matchDetails": [
    {
     "type": "cpe-match",
     "matcher": "stock-matcher",
     "searchedBy": {
      "namespace": "nvd:cpe",
      "cpes": [
       "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
      ]
     },
     "found": {
      "vulnerabilityID": "CVE-2023-24329",
      "versionConstraint": "< 3.11 (unknown)",
      "cpes": [
       "cpe:2.3:a:python:python:*:*:*:*:*:*:*:*"
      ]
     }
    }
   ],
   "artifact": {
    "name": "python",
    "version": "3.10.6",
    "type": "binary",
    "locations": [
     {
      "path": "python3.10"
     }
    ],
    "language": "",
    "licenses": [],
    "cpes": [
     "cpe:2.3:a:python_software_foundation:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*",
     "cpe:2.3:a:python:python:3.10.6:*:*:*:*:*:*:*"
    ],
    "purl": "pkg:generic/python@3.10.6",
    "upstreams": []
   }
  }
 ],
 "source": {
  "type": "directory",
  "target": "/bin"
 },
 "distro": {
  "name": "",
  "version": "",
  "idLike": null
 },
 "descriptor": {
  "name": "grype",
  "version": "0.59.1",
  "configuration": {
   "configPath": "",
   "verbosity": 0,
   "output": "json",
   "file": "",
   "distro": "",
   "add-cpes-if-none": true,
   "output-template-file": "",
   "quiet": false,
   "check-for-app-update": true,
   "only-fixed": false,
   "only-notfixed": false,
   "platform": "",
   "search": {
    "scope": "Squashed",
    "unindexed-archives": false,
    "indexed-archives": true
   },
   "ignore": null,
   "exclude": [],
   "db": {
    "cache-dir": "/home/kali/.cache/grype/db",
    "update-url": "https://toolbox-data.anchore.io/grype/databases/listing.json",
    "ca-cert": "",
    "auto-update": true,
    "validate-by-hash-on-start": false,
    "validate-age": true,
    "max-allowed-built-age": 432000000000000
   },
   "externalSources": {
    "enable": false,
    "maven": {
     "searchUpstreamBySha1": true,
     "baseUrl": "https://search.maven.org/solrsearch/select"
    }
   },
   "match": {
    "java": {
     "using-cpes": true
    },
    "dotnet": {
     "using-cpes": true
    },
    "golang": {
     "using-cpes": true
    },
    "javascript": {
     "using-cpes": true
    },
    "python": {
     "using-cpes": true
    },
    "ruby": {
     "using-cpes": true
    },
    "stock": {
     "using-cpes": true
    }
   },
   "dev": {
    "profile-cpu": false,
    "profile-mem": false
   },
   "fail-on-severity": "",
   "registry": {
    "insecure-skip-tls-verify": false,
    "insecure-use-http": false,
    "auth": []
   },
   "log": {
    "structured": false,
    "level": "warn",
    "file": ""
   },
   "show-suppressed": false,
   "by-cve": false
  },
  "db": {
   "built": "2023-03-22T01:29:27Z",
   "schemaVersion": 5,
   "location": "/home/kali/.cache/grype/db/5",
   "checksum": "sha256:dca4a4a83bc9b1f485991ba187a6d86a33efb06f4819e7303f86124486b5bb3a",
   "error": null
  }
 }
}
